R/fixest-tidiers.R

```diff
@@ -0,0 +1,197 @@
+  ret <- fix_data_frame(coeftable, nn)
+
+  # CIs (if desired) will follow a simiar pattern as above, depending on what
+  # the user specifies in the tidy() call.
+  if (conf.int) {
+    if (has_dots) {
+      CI <- stats::confint(x, level = conf.level, ...)
+      # Bind to rest of tibble
+      colnames(CI) <- c("conf.low", "conf.high")
+      ret <- cbind(ret, unrowname(CI))
+    } else {
+      # It's possible we were passed a summary of a fixest object. If no `...`
+      # arguments were specified, then we want to use the std error that `x`
+      # already has. That is, we don't want to call confint() because
+      # doing so throws away whatever arguments were provided to summary().
+      ci_fact <- abs(qnorm((1 - conf.level) / 2))
+      ret <- mutate(ret,
+        conf.low  = estimate - ci_fact * std.error,
+        conf.high = estimate + ci_fact * std.error
+      )
+    }
+  }
+  as_tibble(ret)
+}
+
+#' @templateVar class fixest
+#' @template title_desc_augment
+#'
+#' @inherit tidy.fixest params examples
+#' @template param_data
+#' @template param_newdata
+#' @param type.predict Passed to [`predict.fixest`][fixest::predict.fixest()]
+#'   `type` argument. Defaults to `"link"` (like `glm.predict`).
+#'
+#' @evalRd return_augment()
+#'
+#' @note Important note: `fixest` models do not include a copy of the input
+#' data, so you must provide it manually.
+#'
+#' augment.fixest only works for [fixest::feols()], [fixest::feglm()], and
+#' [fixest::femlm()] models.
+#'
+#' @export
+#' @family fixest tidiers
+#' @seealso [augment()], [fixest::feglm()], [fixest::femlm()], [fixest::feols()]
+augment.fixest <- function(x, data = NULL, newdata = NULL, type.predict="link", ...) {
+  if (!x$method %in% c("feols", "feglm", "femlm")) {
+    stop("augment is only supported for fixest models estimated with ",
+      "feols, feglm, or femlm\n",
+      "  (supplied model used ", x$method, ")"
+    )
+  }
+
+  if (is.null(newdata)) {
+    df <- data
+  } else {
+    df <- newdata
+  }
+  if (is.null(df)) {
+    stop("Must specify either `data` or `newdata` argument.", call. = FALSE)
+  }
+  df <- as_broom_tibble(df)
+  if (is.null(newdata)) {
+    # use existing data
+    df <- mutate(df,
+      .fitted = predict(x, type = type.predict),
+      .resid = residuals(x)
+    )
+  } else {
+    #
+    df$.fitted <- residuals(x, type = type.predict, newdata = newdata)
+  }
+  df
+}
+
+#' @templateVar class fixest
+#' @template title_desc_glance
+#'
+#' @inherit tidy.fixest params examples
+#'
+#' @note The columns of the result depend on the type of model estimated.
+#'
+#' @return A one-row [tibble::tibble()] with exactly one row and columns:
+#'
+#' - logLik The log-likelihood of the model. [stats::logLik()] may be a useful
+#'   reference.
+#' - AIC Akaike's Information Criterion for the model.
+#' - BIC Bayesian Information Criterion for the model.
+#' - nobs Number of observations used.
+#' - sigma (`feols` models only)
```

Comment:
What do you think always returning the same columns, but the value is just `NA` for non `feols` objects? Does that feel less magical?

Reply:
I don't have strong feelings here. Would you also want to make the `pseudo.r.squared` column's value `NA` for `feols` objects? 
Another option is glossing over the difference between R<sup>2</sup> and pseudo-R<sup>2</sup>, and just returning whichever is appropriate for the model.

Reply:
> Would you also want to make the pseudo.r.squared column's value NA for feols objects?

Yes, let's do this.